test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js

```diff
@@ -77,99 +77,45 @@ const changeGlobalStylesFirstColorPaletteItem = async ( driver, value, pickerOpe
-	);
-
-	await driver.wait(
-		async function () {
-			if (
-				await driverHelper.isElementNotLocated(
-					driver,
-					By.css( '.edit-site-navigation-panel__back-to-dashboard' )
-				)
-			) {
-				await driverHelper.clickWhenClickable(
-					driver,
-					By.css( '.components-navigation__back-button' )
-				);
-				return false;
-			}
-			return true;
-		},
-		config.get( 'explicitWaitMS' ),
-		'Could not reach the "Dashboard" button'
-	);
-
-	await driverHelper.clickWhenClickable(
-		driver,
-		By.css( '.edit-site-navigation-panel__back-to-dashboard' )
-	);
-};
-
-const deleteAll = async function ( driver ) {
-	// Make sure we have posts before trying to delete them
-	const noItems = await driverHelper.isElementLocated( driver, By.css( '#the-list .no-items' ) );
-	if ( noItems ) {
-		return;
-	}
-
-	// Delete all posts
-	await driverHelper.clickWhenClickable( driver, By.css( '#cb-select-all-1' ) );
-	await driverHelper.clickWhenClickable( driver, By.css( '#bulk-action-selector-top' ) );
-	await driverHelper.clickWhenClickable(
-		driver,
-		By.css( '#bulk-action-selector-top option[value="trash"]' )
-	);
-	await driverHelper.clickWhenClickable( driver, By.css( '#doaction' ) );
-
-	// Empty trash
-	await driverHelper.clickWhenClickable( driver, By.css( '.subsubsub .trash a' ) );
-	await driverHelper.clickWhenClickable( driver, By.css( '#delete_all' ) );
-};
-
 const deleteTemplates = async function ( driver ) {
-	const sidebar = await SidebarComponent.Expect( driver );
-	await sidebar.selectTemplates();
-	await deleteAll( driver );
+	await SiteEditorComponent.Expect( driver );
+
+	const getAndDeleteTemplates = async () => {
+		const templates = window.wp.data
+			.select( 'core' )
+			.getEntityRecords( 'postType', 'wp_template', {
+				per_page: -1,
+			} )
+			.filter( ( item ) => item.source === 'custom' );
+		for ( const template of templates ) {
+			await window.wp.data
+				.dispatch( 'core' )
+				.deleteEntityRecord( 'postType', 'wp_template', template.id );
+		}
+	};
+	await driver.executeScript( getAndDeleteTemplates );
 };
 
 const deleteTemplateParts = async function ( driver ) {
-	const sidebar = await SidebarComponent.Expect( driver );
-	await sidebar.selectTemplateParts();
-	await deleteAll( driver );
-};
-
-const backToCalypso = async function ( driver ) {
-	await driverHelper.clickWhenClickable(
-		driver,
-		driverHelper.createTextLocator( By.css( '.wp-menu-name' ), 'Plans' )
-	);
+	await SiteEditorComponent.Expect( driver );
+	const getAndDeleteTemplateParts = async () => {
+		const templateParts = window.wp.data
+			.select( 'core' )
+			.getEntityRecords( 'postType', 'wp_template_part', {
+				per_page: -1,
+			} )
+			.filter( ( item ) => item.source === 'custom' );
+		for ( const templatePart of templateParts ) {
+			await window.wp.data
+				.dispatch( 'core' )
+				.deleteEntityRecord( 'postType', 'wp_template_part', templatePart.id );
+		}
+	};
+	await driver.executeScript( getAndDeleteTemplateParts );
 };
```

Comment:
These two functions are very identical. Maybe we should simplify this as `deleteEntities('wp_template')`?

Reply:
Good idea!  Updated to `deleteCustomEntities` de6d82d